rush-plugins/rush-serve-plugin/src/phasedCommandHandler.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { once } from 'events';
+import https from 'https';
+import type { AddressInfo } from 'net';
+
+import express from 'express';
+
+import { CertificateManager, ICertificate } from '@rushstack/debug-certificate-manager';
+import type {
+  ILogger,
+  RushConfiguration,
+  RushConfigurationProject,
+  RushSession,
+  IPhasedCommand,
+  Operation,
+  ICreateOperationsContext
+} from '@rushstack/rush-sdk';
+
+import { PLUGIN_NAME } from './constants';
+import { IRoutingRule, RushServeConfiguration } from './RushProjectServeConfigFile';
+import type { CommandLineStringParameter } from '@rushstack/ts-command-line';
+
+export interface IPhasedCommandHandlerOptions {
+  rushSession: RushSession;
+  rushConfiguration: RushConfiguration;
+  command: IPhasedCommand;
+  portParameterLongName: string | undefined;
+}
+
+export async function phasedCommandHandler(options: IPhasedCommandHandlerOptions): Promise<void> {
+  const { rushSession, rushConfiguration, command, portParameterLongName } = options;
+
+  const logger: ILogger = rushSession.getLogger(PLUGIN_NAME);
+
+  let activePort: number | undefined;
+
+  command.hooks.createOperations.tapPromise(
+    {
+      name: PLUGIN_NAME,
+      stage: -1
+    },
+    async (operations: Set<Operation>, context: ICreateOperationsContext) => {
+      if (!context.isInitial || !context.isWatch) {
+        return operations;
+      }
+
+      const certManager: CertificateManager = new CertificateManager();
+
+      const certificate: ICertificate = await certManager.ensureCertificateAsync(
+        true, // Allow generating a new certificate
```

Comment:
So we're always running in HTTPS mode? Might make sense to make that configurable.

Reply:
Most browsers make using raw HTTP very difficult nowadays and we have support for HTTPS. I'd be open to adding it as a feature if anyone asks for it, but it adds complexity.